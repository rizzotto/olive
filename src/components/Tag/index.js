import React from 'react';
import useStyles from './styles';
import PropTypes from 'prop-types';

function Tag({
  customStyle,
  disabled,
  isButton,
  name,
  onClick,
  selected,
  ...others
}) {
  const classes = useStyles({ disabled, selected, isButton });

  const Component = isButton ? 'button' : 'div';

  return (
    <Component
      onClick={isButton ? onClick : undefined}
      style={customStyle}
      className={classes.root}
      {...others}
    >
      {name}
    </Component>
  );
}

Tag.propTypes = {
  isButton: PropTypes.bool,
  onClick: PropTypes.func,
  name: PropTypes.string.isRequired,
  selected: PropTypes.bool,
};

Tag.defaultProps = {
  isButton: false,
  onClick: () => {},
  selected: false,
};

export default Tag;
