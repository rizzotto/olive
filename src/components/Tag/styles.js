import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  root: {
    border: props =>
      `1px solid ${
        props.disabled
          ? theme.palette.grey.medium
          : theme.palette.secondary.main
      }`,
    borderRadius: 21,
    padding: '6px 10px',
    background: 'none',
    color: props =>
      props.selected
        ? 'white'
        : props.disabled
        ? 'gray'
        : theme.palette.secondary.main,
    font: 'inherit',
    cursor: props => (props.isButton ? 'pointer' : 'inherit'),
    outline: 'inherit',
    backgroundColor: props =>
      props.selected ? theme.palette.secondary.main : 'transparent',
    transition: '0.3s',

    '&:active': {
      backgroundColor: props => props.isButton && theme.palette.secondary.main,
    },
  },
}));

export default useStyles;
