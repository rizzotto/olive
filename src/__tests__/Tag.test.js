import React from 'react';
import renderer from 'react-test-renderer';
import Tag from '../components/Tag';

test('renders Tag', () => {
  const tree = renderer.create(<Tag name="foo" />).toJSON();
  expect(tree).toMatchSnapshot();
});
