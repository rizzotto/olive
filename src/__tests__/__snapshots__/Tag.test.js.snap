// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tag 1`] = `
<div
  className="makeStyles-root-1 makeStyles-root-2"
>
  foo
</div>
`;
